add pikepdf  and visual c++ dependency

story645 opened the pull request:
@timhoffm noticed that pikepdf is missing from the dependencies list - I have no idea if it's optional or required for testing, but figured getting the ball rolling was a good start. 

Also I can't find any mention of windows dependencies on the page and the error message I got was `Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools` and seemed like a safe requirement to mention. (Also should we document minimum version of C++ needed for linux and os/x?)

## PR Summary

## PR Checklist

<!-- Please mark any checkboxes that do not apply to this PR as [N/A]. -->

- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

<!--
Thank you so much for your PR!  To help us review your contribution, please
consider the following points:

- A development guide is available at https://matplotlib.org/devdocs/devel/index.html.

- Help with git and github is available at
  https://matplotlib.org/devel/gitwash/development_workflow.html.

- Do not create the PR out of main, but out of a separate branch.

- The PR title should summarize the changes, for example "Raise ValueError on
  non-numeric input to set_xlim".  Avoid non-descriptive titles such as
  "Addresses issue #8576".

- The summary should provide at least 1-2 sentences describing the pull request
  in detail (Why is this change required?  What problem does it solve?) and
  link to any relevant issues.

- If you are contributing fixes to docstrings, please pay attention to
  http://matplotlib.org/devel/documenting_mpl.html#formatting.  In particular,
  note the difference between using single backquotes, double backquotes, and
  asterisks in the markup.

We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in.  Please let us know if the reviews are unclear or
the recommended next step seems overly demanding, if you would like help in
addressing a reviewer's comments, or if you have been waiting too long to hear
back on your PR.
-->

tacaswell commented:
I believe this is superseded by #23620.

timhoffm commented:
👍 pikepdf is indeed optional.

pre-commit complains about trailing whitespace. You can self-merge after fixing this.

QuLogic commented:
> Also I can't find any mention of windows dependencies on the page and the error message I got was `Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools` and seemed like a safe requirement to mention. (Also should we document minimum version of C++ needed for linux and os/x?)

When do you get this error?

story645 commented:
> When do you get this error?

It was part of the install when it was trying to do something with freetype and I don't anymore and I'm not sure if it's cause I updated my local repo, updated my c++, or install freetype w/  https://vcpkg.io/en/getting-started.html
